src/gene/query.py

```diff
@@ -423,31 +421,47 @@ def _get_concept_mapping(
             if source == NamespacePrefix.HGNC:
                 source_code = concept_id.upper()
 
+            return Coding(
+                id=concept_id,
+                code=code(source_code),
+                system=NAMESPACE_TO_SYSTEM_URI[source],
+            )
+
+        def _get_concept_mapping(
+            coding_obj: Coding, relation: Relation = Relation.RELATED_MATCH
+        ) -> ConceptMapping:
+            """Get concept mapping for Coding object
+
+            :param coding: A Coding object
+            :param relation: SKOS mapping relationship, default is relatedMatch
+            :return: Concept mapping for identifier
+            """
             return ConceptMapping(
-                coding=Coding(
-                    id=concept_id,
-                    code=code(source_code),
-                    system=NAMESPACE_TO_SYSTEM_URI[source],
-                ),
+                coding=coding_obj,
                 relation=relation,
             )
 
         gene_obj = MappableConcept(
             id=f"normalize.gene.{record['concept_id']}",
-            primaryCode=code(root=record["concept_id"]),
+            primaryCoding=_get_coding_object(record["concept_id"]),
             name=record["symbol"],
             conceptType="Gene",
         )
 
         xrefs = [record["concept_id"], *record.get("xrefs", [])]
         gene_obj.mappings = [
-            _get_concept_mapping(xref_id, relation=Relation.EXACT_MATCH)
+            _get_concept_mapping(
+                _get_coding_object(xref_id), relation=Relation.EXACT_MATCH
+            )
```

Comment:
Since we're always calling `_get_coding_object` with `_get_concept_mapping`, we should make this call inside of `_get_concept_mapping`.